Module#02/api/src/main/java/com/epam/esm/controller/CertificateController.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+    @GetMapping(value = "/search", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Object> search(
+            @RequestParam(value = "name", required = false) final String name,
+            @RequestParam(value = "date", required = false) final Instant date,
+            @RequestParam(value = "tagName", required = false) final String tagName,
+            @RequestParam(value = "description", required = false) final String description,
+            @RequestParam(value = "order", required = false, defaultValue = "UNSORTED") final SortOrder order) {
+        try {
+            Criteria criteria = Criteria.builder()
+                    .tagName(tagName)
+                    .name(name).date(date)
+                    .description(description)
+                    .sortOrder(order).build();
+            return new ResponseEntity<>(
+                    certificateService.getAllBy(criteria),
+                    new HttpHeaders(), OK);
+        } catch (Exception e) {
+            log.error(e.getMessage());
+            return new ResponseEntity<>(NOT_FOUND);
+        }
+    }
+
+    @ResponseBody
+    @PatchMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public final ResponseEntity<Object> update(
+            @PathVariable("id") final Long id,
+            @RequestBody final CertificateDto certificateDto) {
+        try {
+            if (certificateService.getById(id) != null) {
+                certificateDto.setId(id);
+                return new ResponseEntity<>(
+                        certificateService.update(certificateDto),
+                        new HttpHeaders(), OK);
+            }
+            throw new NotFoundException("update", "/certificates/" + id, new HttpHeaders());
+        } catch (Exception e) {
+            log.error(e.getMessage());
+            return new ResponseEntity<>(NOT_FOUND);
+        }
+    }
+
+    @ResponseBody
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public final ResponseEntity<Object> create(
+            @RequestBody final CertificateDto certificateDto) {
+        try {
+            if (certificateService.getByName(certificateDto.getName()) == null) {
+                return new ResponseEntity<>(
+                        certificateService.save(certificateDto),
+                        new HttpHeaders(), OK);
+            } else {
+                throw new CertificateIsExistsException(certificateDto.getName());
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage());
+            return new ResponseEntity<>(NOT_FOUND);
+        }
+    }
+
+    @ResponseBody
+    @DeleteMapping(value = "/{id}")
+    public final ResponseEntity<Void> delete(
+            @PathVariable("id") final Long id) {
+        try {
+            return certificateService.delete(id)
+                    ? new ResponseEntity<>(OK)
+                    : new ResponseEntity<>(NOT_FOUND);
+        } catch (Exception e) {
```

Comment:
ternary operation should not be written in controller layer.

Reply:
appropriate RuntimeException should be thrown

Reply:
Removed all logic to the Service Layer